### ISSUE TITLE
Class balancing with imbalance video classes

### ISSUE THREAD

**Issue by Michelvl92** [NVIDIA/DALI#3047]
Hi,

I use the example as shown in [Video Pipeline Reading Labelled Videos from a Directory](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/sequence_processing/video/video_reader_label_example.html) to train a model with TensorFlow/Keras in a two-class problem. The model is trained in Tensorflow/Keras together with the tf plugin: [nvidia.dali.plugin.tf.DALIDataset](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/plugins/tensorflow_plugin_api.html#nvidia.dali.plugin.tf.DALIDataset) where I pass the dataset to model created with `tf.keras` and use `model.fit` method to train it.

The video dataset consists of 2 classes:
- Class 0: 30 video’s of 6 seconds long
- Class 1: 30 video’s of 120 seconds long

As you can see this dataset has a huge imbalance, for class 1, since it consists of videos that are 20x longer than the class 0 videos, and thus more sequences can be taken by [nvidia.dali.fn.readers.video](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/supported_ops.html#nvidia.dali.fn.readers.video
)

Therefore, I would like for each epoch to randomly pick the same no. of sequences from both classes. Such that every epoch there are the same amount of sequences from both classes, but from class 1 (probably) every time another set is taken.

Is this possible, and if yes, would it be possible to provide an example for TensorFlow/Keras, or show me the right direction in the documentation,

**Comment by JanuszL** [comment]
Hi @Michelvl92,

What you can do is to provide all videos from class 0, 20 times to the reader and then make the epoch size 1/20 of DALI epoch size. DALI random shuffling would make sure that samples (sequences) are randomly taken from the provided dataset definition.
For that, you can use `filenames` and `labels` arguments.

**Comment by Michelvl92** [comment]
Yes, that works, thanks for your clear explanation. Thank you for this useful library, keep up the good work!

**Comment by Michelvl92** [comment]
I just want to leave a comment for anyone, that the best way as I figured out is using class weights. E.g. for TensorFlow/Keras, as explained here: [How to set class weights for imbalanced classes](https://datascience.stackexchange.com/questions/13490/how-to-set-class-weights-for-imbalanced-classes-in-keras)
